src/mlpack/tests/main_tests/nca_test.cpp

```diff
@@ -383,34 +401,52 @@ BOOST_AUTO_TEST_CASE(NCALinearScanTest2)
  */
 BOOST_AUTO_TEST_CASE(NCADifferentNumBasisTest)
 {
-  // Simple dataset.
-  arma::mat x;
-  x.randu(3, 600);
+  // This test can randomly fail and it can be okay, so we run multiple times if
+  // necessary.
+  bool success = false;
+  size_t trial = 0;
+  while (trial < 5)
```

Comment:
I see you used `trial < 3` in the `NCADifferentToleranceTest` test does this one fail more often?

Reply:
I'm not sure---I tried 5 trials initially here.  Personally I would believe that this test is more likely to fail more often, but I'm not sure.  I could always bump the `DifferentToleranceTest` to 5 trials; what do you think?

Reply:
I think either is fine, was just curious if you have seen a higher fail rate on the test.

Reply:
Nope, just an intuitive guess.  Both of the tests did fail if I had no trials though, and the `DifferentToleranceTest` seemed to fail less often (but I didn't write down how often it failed anywhere unfortunately).